### DISCUSSION TITLE
3-fixed  epub

### DISCUSSION THREAD

**Discussion by GEORGEJUNGCOCAINE** [noDRM/DeDRM_tools#683] [General]
It does not work for epubs with the new 3-fixed layout.

Any altenatives? Thnx.

**Comment by ElleKayEm** [comment]
What doesn't work?  Any fixed layout is usually crap for conversion.

**Reply by GEORGEJUNGCOCAINE** [reply]
"This book is locked by DRM"

Any altenatives? 
Thnx.

**Reply by ElleKayEm** [reply]
Where's it from?  What DRM?

**Reply by GEORGEJUNGCOCAINE** [reply]
![image](https://github.com/user-attachments/assets/46cb0eee-ba2b-4cb5-98a3-206450134824)

3-fixed epub from Adobe Digital Editions (ascm)

Here are the epubs: https://gofile.io/d/gp5JJn 

I am not sure if i could find and share the keys from my laptop (on Windows 7)

**Reply by ElleKayEm** [reply]
You download them with ADE authorized with your Adobe ID and on the same computer and user as calibre/DeDRM?

**Reply by GEORGEJUNGCOCAINE** [reply]
I woudn't mind sharing my credentials to the web by PM so that somebody can downnload he epub themself and try. ♥

**Reply by ElleKayEm** [reply]
Not going to mess with my own working setup for your books.  Willing to look at a log of your attempt.

**Reply by GEORGEJUNGCOCAINE** [reply]
Fair enough. Where in my laptop would the log be saved?

**Reply by ElleKayEm** [reply]
Here's how to get a log: https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Reply by GEORGEJUNGCOCAINE** [reply]
Thanks for bearing with me. Here's the log:

```
calibre Debug log
calibre 3.48 Portable embedded-python: True is64bit: False
Windows-7-6.1.7601-SP1 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '7', '6.1.7601')
Python 2.7.15+
Windows: ('7', '6.1.7601', 'SP1', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (10, 0, 9)
calibre 3.48 Portable embedded-python: True is64bit: False
Windows-7-6.1.7601-SP1 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '7', '6.1.7601')
Python 2.7.15+
Windows: ('7', '6.1.7601', 'SP1', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (10, 0, 9)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 71.9842323651 x 71.9822878229
Using calibre Qt style: True
[0.00] Starting up...
[0.02] Showing splash screen...
[0.16] splash screen shown
[0.16] Initializing db...
[0.16] db initialized
[0.16] Constructing main UI...
[1.78] main UI initialized...
[1.78] Hiding splash screen
[16.36] splash screen hidden
[16.36] Started up in 16.36 seconds with 1 books
DeDRM v10.0.9: Trying to decrypt Lazarillo de Tormes (Epub 3 Fijo).epub
DeDRM v10.0.9: Verifying zip archive integrity
DeDRM v10.0.9: Lazarillo de Tormes (Epub 3 Fijo).epub is a secure Adobe Adept ePub for UUID 043e6898-ca67-4b73-a080-f1ee9e018c24
DeDRM v10.0.9: Looking for new default Adobe Digital Editions Keys after 20.7 seconds
DeDRM v10.0.9: Exception when getting default Adobe Key after 20.7 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 565, in ePubDecrypt
  File "calibre_plugins.dedrm.adobekey", line 318, in adeptkeys
NameError: global name 'FileNotFoundError' is not defined
DeDRM v10.0.9: Ultimately failed to decrypt after 20.9 seconds. Read the FAQs at noDRM's repository: https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 1041, in run
  File "calibre_plugins.dedrm.__init__", line 639, in ePubDecrypt
DeDRMError: DeDRM v10.0.9: Ultimately failed to decrypt after 20.9 seconds. Read the FAQs at noDRM's repository: https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md
C:\Users\Usuario\Desktop\calibre\Calibre Portable\Calibre\app\pylib.zip\dateutil\parser.py:601: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
Added Lazarillo de Tormes (Epub 3 Fijo) to db in: 7.2
Added 1 books in 30.8 seconds
Worker Launch took: 0.123999834061
```

**Reply by ElleKayEm** [reply]
Since you are using an older version of calibre because of Windows 7, I suggest trying an older DeDRM 6.8.1: https://github.com/apprenticeharper/DeDRM_tools/releases/tag/v6.8.1

After installing this version, restart calibre before adding your book.

**Reply by JSWolf** [reply]
> Not going to mess with my own working setup for your books. Willing to look at a log of your attempt.

What you can do is setup a portable calibre and copy your settings and use that to try his key and eBooks. That would work to keep your existing calibre in tact.

**Reply by GEORGEJUNGCOCAINE** [reply]
>   trying an older DeDRM 6.8.1: https://github.com/apprenticeharper/DeDRM_tools/releases/tag/v6.8.1

I worked, but the characters of the file are all messed up with those empty squares... https://gofile.io/d/HRnl9c

**Reply by ElleKayEm** [reply]
@JSWolf  Be my guest.  Better for someone to get it working for themselves.

@GEORGEJUNGCOCAINE  I don't know how to resolve that.  At least the DRM is off.

**Reply by GEORGEJUNGCOCAINE** [reply]
@JSWolf @ElleKayEm 

HI

I cannot convert epub with fixed layout. Therefore, I'd like to know is there is some plugin to convert into images that keep the flow of text seen in the reader?

Or maybe I could use a virtual printer? I've tried, but it failed "for memory error" :
https://justpaste.it/dq2y9

Thnx!

**Reply by ElleKayEm** [reply]
Try using 64-bit calibre.  Alternatively, if it is all full page image scans, you could unzip the epub, pull out the image files, and zip them back up to create a cbz file.

**Reply by GEORGEJUNGCOCAINE** [reply]
Cant calibre print batches of specific pages?

**Reply by ElleKayEm** [reply]
Don't think so.  I suggest asking non-DRM ebook questions over at MobileRead.com.  Did you try 64-bit calibre for converting?

**Reply by GEORGEJUNGCOCAINE** [reply]
64-bit? but it would still be the latest version for Windows 7, way before even epub3 fixed layout came to exist.

**Reply by ElleKayEm** [reply]
Ok, I was just thinking it might be a memory issue.

**Reply by GEORGEJUNGCOCAINE** [reply]
64-bit won't convert, and it won't print to pdf either.
BTW, can't calibre convert to images?

**Reply by ElleKayEm** [reply]
I don't know.  Someone at MobileRead might have some suggestion.  Now that the DRM is off, you at least have options.

**Reply by GEORGEJUNGCOCAINE** [reply]
Should I report here though? https://bugs.launchpad.net/calibre
